### PR TITLE
Implementation of categories service for dependency editor

### PR DESCRIPTION
**Pull request by ravsa**

### CODE REVIEW

File: bayesian/utils.py [hunk 1]
```diff
@@ -757,6 +758,49 @@ def get_cve_data(input_json):
...
+        latest_version = '0.0.0'
+    if libio_version in ('-1', ''):
+        libio_version = '0.0.0'
+    if latest_version == '0.0.0' and libio_version == '0.0.0':
+        return return_version
+    libio_version = convert_version_to_proper_semantic(libio_version)
+    latest_version = convert_version_to_proper_semantic(latest_version)
+    try:
+        return_version = libio_version
+        if sv.SpecItem('<' + latest_version).match(sv.Version(libio_version)):
```
Comment (jpopelka) [inline bayesian/utils.py]: How is this different than
`if sv.Version(libio_version) < sv.Version(latest_version):`
?

File: bayesian/api_v1.py [hunk 2]
```diff
@@ -1141,6 +1142,43 @@ def post():
...
+        if 'result' in gremlin_resp and 'requestId' in gremlin_resp:
+            data = gremlin_resp.get('result')
+            if 'data' in data and data['data']:
+                for items in data.get('data'):
+                    category = items.get('category')
+                    package = items.get('package')
+                    if category and package:
+                        pkg_count = category.get('category_deps_count', [0])[0]
+                        _category = category.get('ctname', [''])[0]
+                        pkg_name = package.get('name', [''])[0]
```
Comment (jpopelka) [inline bayesian/api_v1.py]: Do you want to continue even if  `_category` or `pkg_name` are empty ?

File: bayesian/utils.py (hunk 1 above)
Comment (ravsa) [reply bayesian/utils.py]: Actually, I took reference from f8a-server-backbone  https://github.com/fabric8-analytics/f8a-server-backbone/blob/83555309c17ea37203c08cf8c43764f626e71ced/src/utils.py#L167-L168 for this function.

File: bayesian/api_v1.py (hunk 2 above)
Comment (ravsa) [reply bayesian/api_v1.py]: `ctname` and `name` is the category name and package name values for these keys won't be empty if the category and package exist.

**Comment by jpopelka** [comment]
OK, do you want me to merge ?

**Comment by ravsa** [comment]
Yes, please.

**Comment by ravsa** [comment]
[test]

**Pull request merged**